### ISSUE TITLE
Issue with manually bootstrapped Angular app

### ISSUE THREAD

**Issue by andresmatasuarez** [WearyMonkey/ngtemplate-loader#14]
Hi

For the past few days, I have been trying webpack with my current Angular/Express seed app. I got to the point where I was starting to apply production build transformations to the source, such as ngAnnotate and ngTemplates.

So, I got to this loader and installed it and, sadly, have been fighting against it for the past few hours. The loader adds correctly the template entries to the $templateCache service to the webpack bundle, and all is working expectedly.

However, thing is, my Angular app is bootstrapped manually. I need to make an initial request to the server to fetch Application-wide data, and should not be right to bootstrap my app before ensuring I do have that data.

And that's exactly the source of the problem. My manual-bootstrap script is (which is also my webpack entry):

main.coffee
```
'use strict'

angular  = require 'angular'
Dashboard = require './javascripts/app'  # Dashboard = 'dashboard'

# Great source on how to bootstrap an AngularJS app manually:
# https://blog.mariusschulz.com/2014/10/22/asynchronously-bootstrapping-angularjs-applications-with-server-side-data
angular.element(document).ready ->

  app          = angular.module Dashboard
  console.log(app)
  initInjector = angular.injector [ 'ng' ]
  $http        = initInjector.get '$http'

  # Fetch config data
  $http.get '/api/settings'

  # Include data in app
  .then (res) -> app.constant 'Settings', res.data

  # Bootstrap app.
  .then -> angular.bootstrap document, [ 'dashboard' ]
```

`javascripts/app` creates `dashboard` module, set its config and run blocks and exports its name

The `console.log(app)` prints the `dashboard` module object, where I can see that its _runBlocks array has only ONE element. Inspecting the element in the broswer's console I can quickly realize that its one element is the run block from `javascripts/app`.

So, the run blocks added by this loader are not set at this point and the app gets bootstrapped without the templates cache entries. This run blocks are actually set after the app has bootstrapped but they will never be run by that point.

And that's it. I don't know of any hack to make this work without coding any fix. My only suggestion is to augment the loader configuration with a "create" boolean flag that, if set, exposes a new module named %module=% with the cache entries. That way, I can include the module into my app and ensure that the entries are loaded when I need them.

I wanted to discuss this with you before starting to code a PR, so if you have another ideas, here I am ;)

**Comment by WearyMonkey** [comment]
Hi @andresmatasuarez,

I am using ngTemplate with manual bootstrapping in a few projects with no issues, so I don't think that's the problem.

Could you please also share your webpack config? Or better yet, a small sample project?

Cheers!

**Comment by andresmatasuarez** [comment]
Sure

The project is this: https://github.com/andresmatasuarez/angular-express-passport-seed/tree/webpack

I'm migrating from grunt to webpack builds in the branch "webpack". You'll find the webpack.config.js in the root folder.

Please let me know if you need anything else

**Comment by andresmatasuarez** [comment]
Hey, any news on this? have you had any time to check on this?

**Comment by shairez** [comment]
check if you're not trying to require from inside the `templateUrl` property

don't do this:
```
templateUrl; require('./someTemplate.html');
```
instead require from outside the module definition

hope it helps
check if you're not trying to require from inside the `templateUrl` property

don't do this:
```
templateUrl; require('./someTemplate.html');
```
instead require from outside the module definition

hope it helps

**Comment by andresmatasuarez** [comment]
Will try it at home one of these days.

Just for curiosity, why is that not recommended?

**Comment by shairez** [comment]
It creates the `run(function($templateCache)` only at the directive initialization time and not at the actual module run time.

So you fetch the template url before it gets into the cache

**Comment by WearyMonkey** [comment]
@shairez thanks for helping.

@andresmatasuarez any updates?

**Comment by andresmatasuarez** [comment]
Hi there. I have been reviewing this issue. Thanks @shairez for the unconditional help in advance.

I made the changes you guys suggested but I cannot make the thing to add the "angular.module('ng').run([$templaCache......" entries to any of my chunks.

I think they made some improvements, but I am still at it. I pushed the new code to the same branch (webpack) on my seed project. If you guys can review it again, I would be really grateful.
